modules/orientation/src/main/native/android/c/Orientation.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2020 Gluon
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GLUON BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "Orientation.h"
+
+// Graal handles
+static jclass jGraalOrientationClass;
+static jmethodID jGraalNotifyOrientationMethod;
+
+static void initializeGraalHandles(JNIEnv* env) {
+    jGraalOrientationClass = (*env)->NewGlobalRef(env, (*env)->FindClass(env, "com/gluonhq/attach/orientation/impl/AndroidOrientationService"));
+    jGraalNotifyOrientationMethod = (*env)->GetStaticMethodID(env, jGraalOrientationClass, "notifyOrientation", "(Ljava/lang/String;)V");
+}
+
+static void initializeOrientationDalvikHandles() {
```

Comment:
From a quick look, it almost looked to me like this method is no longer required. But it does create an instance of the dalvik orientiation class, to set the lifecycle handler and intent receiver. Perhaps the method should be renamed as it doesn't initialize any class/method/field handles. It actually initializes the dalvik orientation class.

Reply:
The method is required to create the `DalvikOrientationService` instance, by getting its class/method handles only locally. 
I don't see a need to rename it. We use the same convention for all the services (whether we create global handles or not). Imo, it is better to keep it

Reply:
Well, the reason I brought it up, was that I always thought these methods were used to cache some dalvik handles for later use. But it seems it also does some initialization. Which means this method does more then its name is suggesting. Which in turn made me have to look into the method details when I saw it didn't save any of the handles it looked up. Ideally, I would separate the things that load handles for later use from things that do initialization.

Reply:
So you want to split `initialize<ServiceName>DalvikHandles` into something like `initialize<ServiceName>DalvikService` and `initialize<ServiceName>Handles`?
That's for a whole new refactoring...

Reply:
That was the idea yes, at least it is currently hard to see what the method is used for actually.

Reply:
I'll file an issue for a follow-up PR.